.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,65 @@
+name: Emscripten
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [main]
+concurrency:
+  group: ${{ github.workflow }}-${{ github.job }}-${{ github.ref }}
+  cancel-in-progress: true
+defaults:
+  run:
+    shell: bash -e -l {0}
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Emscripten
+            os: ubuntu-24.04
+
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: install mamba
+      uses: mamba-org/setup-micromamba@main
+      with:
+        environment-file: environment-wasm-build.yml
+        init-shell: bash
+        environment-name: sparrow-wasm-build
+
+    - name: Setup default Build Type on *nux
+      if: ${{ runner.os != 'windows' }}
+      run: |
+        echo "ncpus=$(nproc --all)" >> $GITHUB_ENV
+
+    - name: Build sparrow
+      shell: bash -l {0}
+      run: |
+        micromamba create -f environment-wasm-host.yml --platform=emscripten-wasm32
+
+        mkdir build
+        pushd build
+
+        export PREFIX=$MAMBA_ROOT_PREFIX/envs/sparrow-wasm-host
+
+        emcmake cmake \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_FIND_ROOT_PATH=$PREFIX \
+          -DCMAKE_INSTALL_PREFIX=$PREFIX \
+          -DUSE_DATE_POLYFILL=ON \
+          -DBUILD_EXAMPLES=OFF \
+          -DBUILD_TESTS=OFF \
+          -DBUILD_DOCS=OFF \
+          -DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -fPIC" \
+          ..
+
+        emmake make -j ${{ env.ncpus }} install
+
+    # TODO: Implement a way to execute tests possibly through gtest
```

Comment:
it seems we can use this 

http://github.com/man-group/sparrow/pull/130/files#diff-78a7bc1bdd4aae56dc69fba13e39ebb7e6e794ba5e7398020c0696f69f6758f8R69 here.

Shall look into it soon .